notebooks/225-yolov7-optimization/225-yolov7-optimiization.ipynb

```diff
@@ -0,0 +1,820 @@
+    "                                    break\n",
+    "            # Append statistics (correct, conf, pcls, tcls)\n",
+    "            stats.append((correct.cpu(), pred[:, 4].cpu(), pred[:, 5].cpu(), tcls))\n",
+    "    # Compute statistics\n",
+    "    stats = [np.concatenate(x, 0) for x in zip(*stats)]  # to numpy\n",
+    "    if len(stats) and stats[0].any():\n",
+    "        p, r, ap, f1, ap_class = ap_per_class(*stats, plot=True, v5_metric=v5_metric, names=names)\n",
+    "        ap50, ap = ap[:, 0], ap.mean(1)  # AP@0.5, AP@0.5:0.95\n",
+    "        mp, mr, map50, map = p.mean(), r.mean(), ap50.mean(), ap.mean()\n",
+    "        nt = np.bincount(stats[3].astype(np.int64), minlength=nc)  # number of targets per class\n",
+    "    else:\n",
+    "        nt = torch.zeros(1)\n",
+    "    maps = np.zeros(nc) + map\n",
+    "    for i, c in enumerate(ap_class):\n",
+    "        maps[c] = ap[i]\n",
+    "    return mp, mr, map50, map, maps, seen, nt.sum()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Validation function reports following list of accuracy metrics:\n",
+    "\n",
+    "* `Precision` is the degree of exactness of the model in identifying only relevant objects. \n",
+    "* `Recall` measures the ability of the model to detect all ground truths objects.\n",
+    "* `mAP@t` - mean average precision, represented as area under the Precision-Recall curve agregated over all classes in dataset,  where `t` is Intersection Over Union (IOU) threshold, degree of overlaping between ground truth and predicted objects.\n",
+    "            So, `mAP@.5` means that mean average precision calculated at 0.5 IOU threshold, `mAP@.5:.95` - calculated on range IOU threshlods from 0.5 to 0.95 with step 0.05."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mp, mr, map50, map, maps, num_images, labels = test(data=data, model=compiled_model, dataloader=dataloader, names=NAMES)\n",
+    "# Print results\n",
+    "s = ('%20s' + '%12s' * 6) % ('Class', 'Images', 'Labels', 'Precision', 'Recall', 'mAP@.5', 'mAP@.5:.95')\n",
+    "print(s)\n",
+    "pf = '%20s' + '%12i' * 2 + '%12.3g' * 4  # print format\n",
+    "print(pf % ('all', num_images, labels, mp, mr, map50, map))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Optimize model using NNCF Postrainging Quantization API\n",
+    "\n",
+    "[NNCF](https://github.com/openvinotoolkit/nncf) provides a suite of advanced algorithms for Neural Networks inference optimization in OpenVINO with minimal accuracy drop.\n",
+    "We will use 8-bit quantization in post-training mode (without the fine-tuning pipeline) to optimize YOLOv7.\n",
+    "\n",
+    "> **Note**: NNCF Post-training Quantization is available as a preview feature in OpenVINO 2022.3 release.\n",
```

Comment:
Does it mean this part could work after the release of 2022.3? Could it work with 22.2 as well? Thanks~

Reply:
it should works with pot in 2022.2 openvino-dev and NNCF from github if If my understanding correct (I tested it sometime ago, but it was on early POC stage).
@AlexKoff88  is it still true?

Just for clarification, this PR is targeted to `develop` branch for upcoming release, that is why readme provides info about 2022.3